Updates to use Code Editor and other things found while editing (#174)
replicatedhq/kots-field-labs: instruqt/custom-fields/03-updating-the-application/assignment.md

  type: code
  hostname: shell
  path: /home/replicant/
- title: Vendor
  type: website
  url: https://vendor.replicated.com
  new_window: true  
difficulty: basic
timelimit: 600
---

## Setting Up the Dev Environment ##

Before we update the application, let's set up our dev environment. In this lab, you will notice some new tabs:

> * **Dev:** This is a Linux environment preloaded with some handy tools, like the `replicated` command line. This will be our dev enviornment.

Reviewer: ```suggestion
* **Dev:** This is a Linux environment preloaded with some handy tools, like the `replicated` command line. This will be our dev environment.
```

> * **Code Editor:** This is a code editor provided by Instruqt. We will use this edit the manifests.

Reviewer: ```suggestion
* **Code Editor:** This is a code editor provided by Instruqt. We will use this to edit the manifests.
```

The command needs the `SEQUENCE` we want to download, which was displayed in the results of the `release ls` command:
```
SEQUENCE    CREATED                 EDITED                  ACTIVE_CHANNELS
1           2022-09-20T19:53:57Z    0001-01-01T00:00:00Z    stable
```
Let's create a directory structure before we start dowloading files. Create a directory for this lab in your environment and a `manifests` sub directory to store the manifests.

```
mkdir custom-fields-app

cd custom-fields-app

mkdir manifests

```
To download the contents of the release run the following command

```
replicated release download 1 -d ./manifests

```
A succesful download will yield results similar to this:

```shell
  • Fetching Release 1 ✓  
  • Writing files to ./manifests
    • k8s-app.yaml
    • kots-app.yaml
    • kots-support-bundle.yaml
    • nginx-deployment.yaml
    • nginx-feature-off.yaml
    • nginx-service.yaml

```

Now we are ready to update the application.

## Making Updates to the Application

We are going to add a second `ConfigMap` that will be used when the Super Duper Feature is enabled. 

> Head over to the **Code Editor** tab, which contains a director tree that shouls look similar to this

Reviewer: ```suggestion
Head over to the **Code Editor** tab, which contains a directory tree that should look similar to this
```

<p align="center"><img src="../assets/cf-code-edit-init.png" width=450></img></p>

You can expand the **custom-fields-app** directory to see its contents:

<p align="center"><img src="../assets/cf-app-dir.png" width=350></img></p>

Use the **New File** icon (highlighted in red below) to create a new file in the **manifests** directory

<p align="center"><img src="../assets/cf-new-file-icon.png" width=350></img></p>

> Enter `cp-feature-on.yaml` as the file name in the dialog 

Reviewer: Should this be `nginx-featue-on.yaml` like in the screenshot?

Author: Yes and have made a new commit to reflect the correct file name

The above basically states that if the Custom License Field is set to true, the value of the `name` key is `nginx-feature-on`, otherwise the value will be `nginx-feature-off`

To save changes, click on the save icon on the file tab in the editor. To create a new release head back to the **Dev** tab and run the following command (make sure you are on the `custom-fields-app` directory):

```
replicated release create --version 0.1.1 --release-notes "Update for Super Duper feature" \
  --promote Stable --yaml-dir manifests
```

The output should look very similar to the output below

<p align="center"><img src="../assets/cf-new-release.png" width=450></img></p>

> Let's verify our release we indeed created and promoted to the channel by running `replicated release ls`. There should ne a new SEQUENCE associated to the **CustomFields** channel.

Reviewer: ```suggestion
Let's verify our release was indeed created and promoted to the channel by running `replicated release ls`. There should be a new SEQUENCE associated to the **CustomFields** channel.
```
